security/monitoring.adoc

```diff
@@ -0,0 +1,202 @@
+(where entries can easily be identified by searching for *"AUDIT"*).
+
+xref:../install_config/master_node_configuration.adoc#master-node-config-audit-config[Audit
+log entries] consist of one line recording each REST request when it is
+received and one line with the HTTP response code when it completes. For
+example, here is a record of the system administrator requesting a list
+of nodes:
+
+[source, bash]
+----
+2017-10-17T13:12:17.635085787Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" ip="192.168.122.156" method="GET" user="system:admin" groups="\"system:cluster-admins\",\"system:authenticated\"" as="<self>" asgroups="<lookup>" namespace="<none>" uri="/api/v1/nodes"
+2017-10-17T13:12:17.636081056Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" response="200"
+----
+
+For this example, it might be useful to poll the log periodically by using the following
+command to see if the number of requests failed:
+
+[source, bash]
+----
+$ tail -5000 /var/log/openshift-audit.log \
+  | grep -Po 'response="..."' \
+  | sort | uniq -c | sort -rn
+
+   3288 response="200"
+      8 response="404"
+      6 response="201"
+----
+
+Most valid requests succeed with a `200` or `201` response code. `401` and
+`403` are simple redirects, and `404` is typically a benign request for a
+resource that does not exist. Return code `400` might be of
+interest as it indicates a malformed request, which should not occur with
+most clients. Responses in the `500` range indicate server errors, which
+might be a result of bugs, system failures, or even malicious activity.
+If an unusual number of error responses are found, the audit log entries
+for corresponding requests can be retrieved for further investigation.
+
+It can also be useful to look for unusual numbers of requests by a
+particular user or group. Note that the IP address of the request is typically
+a cluster host or API load balancer, and there is no record of the IP address
+behind a load balancer proxy request (however, load balancer logs might
+potentially be useful for determining request origin). The following
+example lists the top 10 users by number of requests in the last 5000
+lines of the audit log:
+
+[source, bash]
+----
+$ tail -5000 /var/log/openshift-audit.log \
+  | grep -Po ' user="(.*?)(?<!\\)"' \
+  | sort | uniq -c | sort -rn | head -10
+
+    976  user="system:openshift-master"
+    270  user="system:node:origin-node-1.example.local"
+    270  user="system:node:origin-master.example.local"
+     66  user="system:anonymous"
+     32  user="system:serviceaccount:kube-system:cronjob-controller"
+     24  user="system:serviceaccount:kube-system:pod-garbage-collector"
+     18  user="system:serviceaccount:kube-system:endpoint-controller"
+     14  user="system:serviceaccount:openshift-infra:serviceaccount-pull-secrets-controller"
+     11  user="test user"
+      4  user="test \" user"
+----
+
+More sophisticated queries generally require more sophisticated
```

Comment:
Is it necessary to say this? If so - think of a synonym for sophisticated for one of the two uses of the word.